api/resolvers/search.js

```diff
@@ -511,21 +505,13 @@ export default {
       })).map((item, i) => {
         const e = sitems.body.hits.hits[i]
 
-        if (e.highlight?.['title.exact']?.[0]) {
-          item.searchTitle = e.highlight['title.exact'][0]
-        } else if (e.highlight?.title?.[0]) {
-          item.searchTitle = e.highlight.title[0]
-        } else {
-          item.searchTitle = item.title
-        }
+        // prefer the fuzzier highlight for title
+        item.searchTitle = e.highlight?.title?.[0] || e.highlight?.['title.exact']?.[0] || item.title
+
+        // prefer the exact highlight for text
+        const searchTextHighlight = [...(e.highlight?.['text.exact'] || []), ...(e.highlight?.text || [])]
+        item.searchText = searchTextHighlight?.slice(0, 5)?.join(' ... ')
```

Comment:
If the title is present and the exact title matches, it means the non-exact title has those matches but it will also have the non-exact matches ... it's like a superset of matches, so I changed it to prefer the non-exact highlights if present on the title.

For the text, we prefer the exact matches, but if there are <5, we also want the non-exact matches, so I concat'd them together then sliced them.